t/apicast-policy-rewrite-url-captures.t

```diff
@@ -0,0 +1,187 @@
+use lib 't';
+use Test::APIcast::Blackbox 'no_plan';
+
+run_tests();
+
+__DATA__
+
+=== TEST 1: one transformation that matches
+--- configuration
+{
+  "services": [
+    {
+      "id": 42,
+      "backend_version":  1,
+      "backend_authentication_type": "service_token",
+      "backend_authentication_value": "token-value",
+      "proxy": {
+        "api_backend": "http://test:$TEST_NGINX_SERVER_PORT/",
+        "proxy_rules": [
+          { "pattern": "/", "http_method": "GET", "metric_system_name": "hits", "delta": 2 }
+        ],
+        "policy_chain": [
+          { "name": "apicast.policy.apicast" },
```

Comment:
I'll change these tests so they use the Echo policy instead. I think that will simplify the code.

I'll keep just one with the APIcast policy just to make sure that there are not any problems when combining it with the new policy.

Reply:
I'd love that 👍

Reply:
I just realized that we can't do that because the Echo policy uses `ngx.say(ngx.var.request)`, so it just works with the original request.

Reply:
So the upstream policy pointing to the echo endpoint? That is still simpler than APIcast + dependencies.

Reply:
Using the echo policy requires more work because we are only interested in checking the path and the query args, but the echo returns more data.

In the end, I opted for deleting the APIcast policy from the chain and adding an upstream policy that redirects all the request to the upstream defining in the config. I think this is the simplest solution.

I kept one of the tests I had with the APIcast policy just to make sure that both policies can be combined without any problems.